src/views/decodeDynamic/index.tsx

```diff
@@ -0,0 +1,354 @@
+import {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import { DecoderDynamicParams } from '@components/callParam/decoderDynamicParam';
+import { QueryButton } from '@components/callParam/queryButton';
+import { QueryFormContainer } from '@components/callParam/queryFormContainer';
+import { QueryResult } from '@components/callParam/queryResult';
+import { QueryResultContainer } from '@components/callParam/queryResultContainer';
+import { QueryViewContainer } from '@components/callParam/queryViewContainer';
+import { Loader } from '@components/loader';
+import { PDSelect } from '@components/pdSelect';
+import { useStoreChain } from '@stores';
+import { useDynamicBuilder } from 'src/hooks/useDynamicBuilder';
+
+import type { EnumVar } from '@polkadot-api/metadata-builders';
+
+const decoderTypes = [
+  'Storage',
+  'Runtime',
+  'Extrinsic',
+];
+
+const decoderTypeSelectItems = decoderTypes.map((decoder) => ({
+  key: `decoder-${decoder}`,
+  label: decoder,
+  value: decoder,
+}));
+
+interface IQuery {
+  type: string;
+  pallet: string;
+  method: string;
+  id: string;
+  args: string;
+}
+
+const DecoderDynamic = () => {
+  const chain = useStoreChain?.use?.chain?.();
+  const metadata = useStoreChain?.use?.metadata?.();
+  const lookup = useStoreChain?.use?.lookup?.();
+  const dynamicBuilder = useDynamicBuilder();
+
+  const [
+    queries,
+    setQueries,
+  ] = useState<IQuery[]>([]);
+  const [
+    callParams,
+    setCallParams,
+  ] = useState<string>('');
+  const [
+    decoderType,
+    setDecoderType,
+  ] = useState<string>(decoderTypes[0]);
+  const [
+    pallet,
+    setPallet,
+  ] = useState<string>('');
+  const [
+    method,
+    setMethod,
+  ] = useState<string>('');
+
+  const palletSelectItems = useMemo(() => {
```

Comment:
Could we check if we can combine  palletSelectItems and methodSelectItems somehow